chi2/chinotype.py

```diff
@@ -593,14 +670,15 @@ def dbtrx():
 
 
     def getFilterSql(self):
-        sql = 'select c_name from {0}.schemes'.format(self.metaschema)
+        sql = 'select c_name from {0}'.format(self.chischemes)
         if 'ALL' in self.filter:
-            sql += '\nwhere 1=1'
+            sql += ' where 1=1'
         elif len(self.filter) > 0:
-            sql += '\nwhere 1=0'
-        for p in range(0, len(self.filter)):
-            if self.filter[p] != 'ALL':
-                sql += '\nor c_key like :{0} || \'%\''.format(p)
+            sql += ' where 1=0'
+        #for p in range(0, len(self.filter)):
+        for p in self.filter:
+            if p != 'ALL':
+                sql += ' or c_key like \'{0}\' || \'%\''.format(p)
```

Comment:
This was not a bug. The functionality works as intended, e.g.

$ python chinotype.py -m 2050 -f "%DX" -f CPT -o -v

The function getFilterSql was intentionally using the subscript of the filter array, no the string itself. This builds a SQL statement with parameter values like :0, :1, ... which are passed into do_log_sql as the 3rd argument.